sdk/core/Azure.Core/src/Pipeline/LoggingPolicy.cs

```diff
@@ -92,19 +98,19 @@ private static async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemo
 
             var textResponse = ContentTypeUtilities.TryGetTextEncoding(message.Response.Headers.ContentType, out Encoding? responseTextEncoding);
 
-            bool wrapResponseStream = message.Response.ContentStream != null && message.Response.ContentStream?.CanSeek == false && s_eventSource.ShouldLogContent(isError);
+            bool logResponseContent = _logContent && message.Response.ContentStream != null;
+            bool wrapResponseContent = logResponseContent && message.Response.ContentStream?.CanSeek == false && s_eventSource.ShouldLogContent(isError);
```

Comment:
You can check the generated IL, but at least when targeting net45 my old code base had to change a lot of this syntax not to use the Elvis operator when comparing against value types. It causes extra allocations. Just separate check if `ContentStream != null` prior.

In a frequent code path, this could add up (it did in our case).

Reply:
I checked, the Elvis doesn't result in allocation and produces almost identical IL and asm